src/ui/popupmenu/completion_item_widget.rs

```diff
@@ -37,11 +40,21 @@ impl CompletionItemWidgetWrap {
         let grid = gtk::Grid::new();
         grid.set_column_spacing(10);
 
-        let buf = get_icon_pixbuf(&item.kind.as_str(), icon_fg, size);
-        let kind = gtk::Image::new_from_pixbuf(&buf);
-        kind.set_tooltip_text(format!("kind: '{}'", item.kind).as_str());
-        kind.set_margin_start(margin);
-        grid.attach(&kind, 0, 0, 1, 1);
+        let buf = get_icon_pixbuf(&item.kind, icon_fg, size);
+        let image = match &item.kind {
+            CompletionItemKind::Unknown => {
+                // If another item in the popupmenu has a known kind,
+                // give kind_image an icon.
+                if items.iter().any(|item| !item.kind.is_unknown()) {
```

Comment:
This is being ran multiple times while it can be assigned to a variable just once, perhaps before line 44